server/src/main/java/com/cloud/acl/DomainChecker.java

```diff
@@ -199,6 +200,9 @@ public boolean checkAccess(Account caller, ControlledEntity entity, AccessType a
     private boolean checkOperationPermitted(Account caller, ControlledEntity entity) {
         User user = CallContext.current().getCallingUser();
         Project project = projectDao.findByProjectAccountId(entity.getAccountId());
+        if (project == null) {
```

Comment:
Are there usage of this method where throwing an exception would break flow? (or, are we expecting all usages of this method to be project related?)

Reply:
this method is called only if the entity is owned by a project - which would mean that ideally project should be present if the entity's account is of Project type